cmd/sts-handlers.go

```diff
@@ -493,6 +493,33 @@ func (sts *stsAPIHandlers) AssumeRoleWithSSO(w http.ResponseWriter, r *http.Requ
 		cred.ParentUser = base64.RawURLEncoding.EncodeToString(bs)
 	}
 
+	// Deny this assume role request if the policy that the user intends to bind
+	// has a sts:DurationSeconds condition, which is not satisfied as well
+	{
+		policies := policyName
+		if policies == "" {
+			var err error
+			_, policies, err = globalIAMSys.GetRolePolicy(roleArnStr)
+			if err != nil {
+				writeSTSErrorResponse(ctx, w, ErrSTSAccessDenied, errors.New("this user does not have enough permission"))
+				return
+			}
+		}
+		for _, st := range globalIAMSys.GetCombinedPolicy(policies).Statements {
+			if st.Actions.Contains(policy.AssumeRoleWithWebIdentityAction) {
+				if !st.IsAllowed(policy.Args{
+					Action:          policy.AssumeRoleWithWebIdentityAction,
+					ConditionValues: getSTSConditionValues(r, "", cred),
+					IsOwner:         false,
+					Claims:          cred.Claims,
```

Comment:
We have something called as 

```
                DenyOnly:        checkDenyOnly,
```

why not use that?

Reply:
why do you want it ?

Reply:
Because I want to avoid writing custom code to check for STS policies.